magenta-lib/src/main/scala/magenta/tasks/SetStackPolicyTask.scala

```diff
@@ -42,7 +42,12 @@ object StackPolicy {
     "AWS::ApiGateway::RestApi",
     "AWS::ApiGateway::DomainName",
     // buckets (although we think they can't be deleted with content)
-    "AWS::S3::Bucket"
+    "AWS::S3::Bucket",
+    // DNS infrastructure
+    "Guardian::DNS::RecordSet",
```

Comment:
My original intention with this PR was just to add `Guardian::DNS::RecordSet`, but as I was here I thought it was worth protecting some `Route53` resources too )although I suspect that these are not CloudFormed very often at the moment anyway)